### ISSUE TITLE
How to obtain IP-EOM-CCSD amplitudes

### ISSUE THREAD

**Issue by hebrewsnabla** [pyscf/pyscf#1136]
I tried IP-EOM-CCSD in example and can directly obtain the energy of ionized states. But how can I get amplitudes (like ORCA output as follows)?
```
IROOT=  1:  0.571735 au    15.558 eV  125481.3 cm**-1
  Amplitude    Excitation
  -0.694069     4 -> x
```

**Comment by tberkel** [comment]
Hi, The IP-EOM-CCSD kernel returns the energies and (right-hand) eigenvectors, which are also saved in `eom.v`. You can then do `r1, r2 = eom.vector_to_amplitudes(v)`, which have storage like `r[i,a]` and `r[i,j,a,b]` with the MO indices sorted as usual (technically, as ordered in `mf.mo_coeff`). At the moment, we don't have pretty printing (e.g. like `analyze()` elsewhere) but it would be good to add in the future.

**Comment by tberkel** [comment]
Hm, after some quick testing, I'm not sure. If you additionally freeze the
core in PySCF, which is the default in Orca, you get much better agreement
in the eigenvalues, so clearly they are finding the same roots. I think
Orca also prints the "percent singles character", which PySCF calls "qpwt"
-- do those agree? The PySCF amplitudes seem to suggest a lot more "singles
character" in roots 1 and 4 than Orca ...

**Comment by hebrewsnabla** [comment]
Yeah, I get much better agreement in the eigenvalues after applying frozencore in PySCF.
There's only very slight change in the r1.

The qpwt of PySCF result is 0.945 0.485 0.485 0.903, while the percentage singles character of ORCA result is 93.12 95.99 95.99 89.20. 
It looks like 0.485 is half of 0.9599 approximately. So, maybe PySCF and ORCA have different definition on singles character and amplitude?

**Comment by pwborthwick** [comment]
If I understand this correctly PySCF eigenvectors for EOM are in a form that reduces the large space. Hence (nocc, nocc, nvir) because of the anti-symmetric nature of the space is used as (np.tril_indices(nocc, k=-1), nvir). The vectors_to_amplitudes functions then restore the full dimensions of the large space re-shaping for IP to r1(nocc) and r2(nocc, nocc, nvir). The eigenvectors as returned by the Davidson iterations are normalixed, vector_to_amplituding those vectors by adding new elements destroys the normalization. If you do
```
for i in range(4):
    r1, r2 = cc.eom_rccsd.vector_to_amplitudes_ip(np.array(cip[i]), mycc.nmo, mycc.nocc)
    v = np.concatenate((r1.ravel(), r2.ravel()))
    v /= np.linalg.norm(v)

    r1, r2 = v[:mycc.nocc], v[mycc.nocc:]
    u = r1.ravel()
    print('QPWT= ', np.linalg.norm(u)**2)
```
you will get
```
QPWT=  0.9456507360530719
QPWT=  0.9704506744732773
QPWT=  0.9704506744732773
QPWT=  0.9043395312626037
```
which are at least qualitatively in agreement with ORCA.

**Comment by hebrewsnabla** [comment]
Thanks @pwborthwick !

**Comment by verena-neufeld** [comment]
Hi @hebrewsnabla , has this been answered?
